### DISCUSSION TITLE
After running "remesh_smooth", the new mesh did not save the vertex colors of the old mesh.

### DISCUSSION THREAD

**Discussion by skylinestars** [BrunoLevy/geogram#176] [Q&A] [answered]
Excuse me，After running "remesh_smooth", the new mesh did not save the vertex colors of the old mesh. How should I calculate the vertex colors of the new mesh.

**Comment by BrunoLevy** [comment, marked as answer]
Yes, the result of `remesh smooth` is a mesh that is completely different from the original mesh, so it does not have the initial attributes. You will have to transfer the attributes, which you can do by creating an AABB (axis-aligned bounding box tree) and find the nearest triangle of the initial mesh for each vertex of the remesh, then interpolate the colors from the mesh. Examples of such attribute transfers are in [`geogram/mesh/mesh_baking.cpp`](https://github.com/BrunoLevy/geogram/blob/main/src/lib/geogram/mesh/mesh_baking.cpp)
